storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -172,8 +186,15 @@ impl StateMerklePruner {
                     batch.delete::<StaleNodeIndexSchema>(&index)
                 })?;
 
-                // Delete the stale node indices.
-                self.db.write_schemas(batch)?;
+                let ledger_batch = SchemaBatch::new();
+                ledger_batch.put::<PrunerMinReadableVersionIndexSchema>(
+                    &PrunerTag::StateMerklePruner,
+                    &PrunedUtilVersion::LatestVersion(new_min_readable_version),
+                )?;
+
+                // Commit to DB.
+                self.ledger_db.write_schemas(ledger_batch)?;
```

Comment:
1. We don't need a batch for the ledger_db if you only write one row. `self.ledger_db.put::<xSchema>(&k, &v)`
2. we should write to state_merkle_db first and then persist the progress? Otherwise on interruption it's more likely we miss things to be pruned. Actually to make sure of consistency, I'd even go even further to check if there's index items before where you assumed, by making a rev_iter.

Reply:
To resolve the issue, write the metadata of state pruner to the state db to guarantee atomicity.